maven-model-builder/src/main/java/org/apache/maven/utils/Os.java

```diff
@@ -0,0 +1,213 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.utils;
+
+import java.util.Locale;
+import java.util.stream.Stream;
+
+/**
+ * OS support
+ */
+public class Os {
+
+    /**
+     * The OS Name.
+     */
+    public static final String OS_NAME = System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
+
+    /**
+     * The OA architecture.
+     */
+    public static final String OS_ARCH = System.getProperty("os.arch").toLowerCase(Locale.ENGLISH);
+
+    /**
+     * The OS version.
+     */
+    public static final String OS_VERSION = System.getProperty("os.version").toLowerCase(Locale.ENGLISH);
+
+    /**
+     * OS Family
+     */
+    public static final String OS_FAMILY;
+
+    public static final boolean IS_WINDOWS;
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_WINDOWS = "windows";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_WIN9X = "win9x";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    public static final String FAMILY_NT = "winnt";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_OS2 = "os/2";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_NETWARE = "netware";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_DOS = "dos";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_MAC = "mac";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_TANDEM = "tandem";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_UNIX = "unix";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_OPENVMS = "openvms";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_ZOS = "z/os";
+
+    /**
+     * OS family that can be tested for. {@value}
+     */
+    private static final String FAMILY_OS400 = "os/400";
+
+    /**
+     * OpenJDK is reported to call MacOS X "Darwin"
+     *
+     * @see <a href="https://issues.apache.org/bugzilla/show_bug.cgi?id=44889">bugzilla issue</a>
+     * @see <a href="https://issues.apache.org/jira/browse/HADOOP-3318">HADOOP-3318</a>
+     */
+    private static final String DARWIN = "darwin";
+
+    /**
+     * The path separator.
+     */
+    private static final String PATH_SEP = System.getProperty("path.separator");
+
+    static {
```

Comment:
Please add a comment explaining why this is a static initializer instead of initializing the fields directly. I see it now, but I almost missed it; and I can believe someone down the road might remove this. 

Although now that I look even closer, I wonder if there's a problem since this calls a static method before the static initializer has completed. Maybe replace the call to isWindowsFamily with OS_NAME.contains(FAMILY_WINDOWS) instead.

Reply:
> Please add a comment explaining why this is a static initializer instead of initializing the fields directly. I see it now, but I almost missed it; and I can believe someone down the road might remove this.

It's not possible to remove the initializer of a static final constant, the compiler would reject with an error.  Anyway, I've added a comment.

> Although now that I look even closer, I wonder if there's a problem since this calls a static method before the static initializer has completed. Maybe replace the call to isWindowsFamily with OS_NAME.contains(FAMILY_WINDOWS) instead.

That's what it ends up doing [in the `isWindowsFamily` method](https://github.com/apache/maven/pull/1249/files/f4eefa52d68b4ce61026030ac0a27e57346efe5f#diff-0d69abd148e63b11a4c58601c7e7ae73fda92b61a9facc97333061be7bec1ee8R141) btw.  I'd rather avoir duplicating the logic, and the `isWindowsFamily` is the one used by the OS activation.